propagator/jaeger/lib/opentelemetry/propagator/jaeger/text_map_extractor.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'cgi'
+
+# OpenTelemetry is an open source observability framework, providing a
+# general-purpose API, SDK, and related tools required for the instrumentation
+# of cloud-native software, frameworks, and libraries.
+#
+# The OpenTelemetry module provides global accessors for telemetry objects.
+# See the documentation for the `opentelemetry-api` gem for details.
+module OpenTelemetry
+  # Namespace for OpenTelemetry propagator extension libraries
+  module Propagator
+    # Namespace for OpenTelemetry Jaeger propagation
+    module Jaeger
+      # Extracts context from carriers
+      class TextMapExtractor
+        TRACE_SPAN_IDENTITY_REGEX = /\A(?<trace_id>(?:[0-9a-f]){1,32}):(?<span_id>([0-9a-f]){1,16}):[0-9a-f]{1,16}:(?<sampling_flags>[0-9a-f]{1,2})\z/.freeze
+        ZERO_ID_REGEX = /^0+$/.freeze
+
+        # Returns a new TextMapExtractor that extracts Jaeger context using the
+        # specified header keys
+        #
+        # @param [optional Getter] default_getter The default getter used to read
+        #   headers from a carrier during extract. Defaults to a
+        #   {OpenTelemetry::Context:Propagation::TextMapGetter} instance.
+        # @return [TextMapExtractor]
+        def initialize(default_getter = Context::Propagation.text_map_getter)
+          @default_getter = default_getter
+        end
+
+        # Extract Jaeger context from the supplied carrier and set the active
+        # span in the given context. The original context will be return if
+        # Jaeger cannot be extracted from the carrier.
+        #
+        # @param [Carrier] carrier The carrier to get the header from.
+        # @param [Context] context The context to be updated with extracted
+        #   context
+        # @param [optional Callable] getter An optional callable that takes a
+        #   carrier and a key and returns the value associated with the key. If
+        #   omitted the default getter will be used which expects the carrier to
+        #   respond to [] and []=.
+        # @yield [Carrier, String] if an optional getter is provided, extract
```

Comment:
Docs are out of date. The getter is no longer a block. Take a look at the docs for other extractors.